### ISSUE TITLE
Add OAuth for users to authenticate

### ISSUE THREAD

**Issue by yashrsharma44** [yashrsharma44/grpc-chat-app#5]
Acceptance Criteria - 
* Add OAuth for users
* OAuth support for GitHub / Google

**Comment by harshil15999** [comment]
@yashrsharma44  is it wise to use Oauth since it is a terminal based chatroom? shouldnt we use ssh

**Comment by yashrsharma44** [comment]
Yeah, that's true. Originally, I intended to use this application in a web browser though gRPC gateway - https://github.com/grpc-ecosystem/grpc-gateway

But I later added that we can add in a terminal support as well.

For this issue we shall use SSH for terminal app and oauth for web app as well.
Since webapp hasn't been made, we can concentrate on SSH to start with.
